### PR TITLE
Maintenance/release process

### PR DESCRIPTION
**Pull request by colmsnowplow**
PR to update release process and tweak the testing GH action config

### CODE REVIEW

File: .github/workflows/prerelease.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  deploy:
+    runs-on: ubuntu-20.04
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Validate input
+        run: |
+          if ! [[ ${{ github.event.inputs.version }} =~ v[0-9]*\.[0-9]*\.[0-9]*-.* ]]; then
```
Comment (adatzer) [inline .github/workflows/prerelease.yaml:20]: this matches also `vX.X.X-`. The last `*` could be replaced with `+` if needs be stricter.

File: VERSION [hunk 2]
```diff
@@ -0,0 +1 @@
+0.2.2
```
Comment (adatzer) [inline VERSION:1]: Will this be bumped?

**Review by TiganeteaRobert — APPROVED** [review]
LGTM!

**Review by adatzer — COMMENTED** [review]
Looks great!
To confirm:

1. For the `prerelease` workflow, the branch will be selected from the drop-down?
2. What if someone pushes a tag as usual? would it make sense to trigger on `push` for tags instead of `workflow_dispatch` ? 

[this action](https://github.com/softprops/action-gh-release) is also nice if we want to automate the github release as well!

File: .github/workflows/prerelease.yaml (hunk 1 above)
Comment (colmsnowplow) [reply .github/workflows/prerelease.yaml:20]: nice catch, thank you!

File: VERSION (hunk 2 above)
Comment (colmsnowplow) [reply VERSION:1]: Current release is 0.2.2 (0.2.3 tag was pushed in error). With 0.3.0 we will bump it yes.

**Comment by colmsnowplow** [comment]
> For the prerelease workflow, the branch will be selected from the drop-down?

For both the branch will be selected from drop down. But `release` fails if it's not master (and if the tag's not a full release).

> What if someone pushes a tag as usual? would it make sense to trigger on push for tags instead of workflow_dispatch ?

Annoyingly I couldn't find a good option to prevent this - protected tags might work but we'd need to work out details. In the meantime, best we can do is just ask people not to manually push tags, and to use the workflow_dispatch instead. If they do so, it should be far more difficult to make a mistake.

**Comment by colmsnowplow** [comment]
@adatzer Patched all regexes to use `+` instead of `*` - well spotted, is better this way as otherwise `v0..1` would also have passed.

**Review by adatzer — APPROVED** [review]

**Pull request merged**